api/app/controllers/assessments.js

```diff
@@ -14,6 +18,24 @@ module.exports = {
         .then((assessment) => reply(assessmentSerializer.serialize(assessment)).code(201))
         .catch((error) => reply(Boom.badImplementation(error)));
     }
+  },
+
+  getNextChallenge: {
+    handler: (request, reply) => {
+
+      assessmentRepository
+        .get(request.params.id)
+        .then((assessment) => {
+          assessmentService.getAssessmentNextChallengeId(assessment, null)
+          .then((challenge_id) => {
+            challengeRepository
+              .get(challenge_id)
+              .then((challenge) => reply(challengeSerializer.serialize(challenge)))
+          })
+          .catch((error) => reply(Boom.badImplementation(error)));
```

Comment:
Le truc merveilleux avec les promesses, c'est qu'[il suffit d'une seule instruction `catch` en fin (ou en cours) de chaîne](http://odetocode.com/blogs/scott/archive/2015/10/01/javascript-promises-and-error-handling.aspx). Ex : 
```javascript
doWork1()
    .then(doWork2)
    .then(doWork3) 
    .then(doWork4)
    .catch(errorHandler);
```